app/main.py

```diff
@@ -1,52 +1,126 @@
 import json
-import xml.etree.ElementTree as ET
+import xml.etree.ElementTree as ElementTree
+from abc import ABC, abstractmethod
```

Comment:
you should create a few files for serializers, displayers, etc, so that it's easier to manage them